x-pack/plugins/security_solution/public/flyout/document_details/shared/hooks/use_fetch_graph_data.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import type {
+  GraphRequest,
+  GraphResponse,
+} from '@kbn/cloud-security-posture-common/types/graph/latest';
+import { useHttp } from '../../../../common/lib/kibana';
+
+export const useFetchGraphData = (req: GraphRequest, options?: { enabled?: boolean }) => {
```

Comment:
can you add some docs here?

Reply:
and also a `UseFetchGraphDataParams` and `UseFetchGraphDataResult` interfaces with docs for each prop?